### ISSUE TITLE
[Bug]: OpenPI PPO critic loss silently broadcasts [B] values against [B, 1] returns

### ISSUE THREAD

**Issue by anHappyDog** [RLinf/RLinf#1568]
### Affected component

critic when using ppo.

### Bug description

In the OpenPI RL path, rollout values and returns have shape [B, 1], while the value head output produced during PPO recomputation has shape [B]. The critic loss directly computes `returns - values` and `values - prev_values`, causing PyTorch to silently broadcast these tensors to [B, B].

As a result, each value prediction is trained against every return in the batch instead of only its corresponding return. This produces incorrect critic gradients, encourages predictions to collapse toward the batch-level central return, corrupts value-clipping statistics, and creates an unnecessary O(B²) intermediate tensor.

The value output should be standardized to [B, 1], and the critic loss should assert that `values`, `returns`, and `prev_values` have identical shapes. A regression test with B > 1 and distinct return values should also be added.

### Configuration YAML file

like libero_spatial_ppo_openpi

### Log file

no logs.

### Environment

python -V:
uv pip list:
nvidia-smi:
RLinf version:
Docker image tag: 

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/resources/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by kakiuwang-ui** [comment]
I tried to reproduce this and could not — the residuals never broadcast, because the embodied actor does not call `compute_ppo_critic_loss` directly.

`policy_loss()` routes every embodied loss through `preprocess_loss_inputs()` first (`rlinf/algorithms/registry.py:85-86`). For `reward_type: chunk_level`, that function flattens `values`, `prev_values` and `returns` (`rlinf/algorithms/utils.py:298-309`), then `expand_to_target_dim` re-expands all three against `logprobs.shape` identically. A `[B]` value tensor and a `[B, 1]` one land on the same shape, so the `[B]` produced by `default_forward` is erased before the loss sees it.

Run against `main` with the shapes from `libero_spatial_ppo_openpi.yaml` (`reward_type: chunk_level`, `logprob_type: chunk_level`, B=4):

```python
kw = dict(
    task_type="embodied", loss_type="actor_critic",
    logprob_type="chunk_level", reward_type="chunk_level", single_action_dim=7,
    logprobs=torch.zeros(4, 8, 7), old_logprobs=torch.zeros(4, 8, 7),
    advantages=torch.zeros(4, 1), loss_mask=torch.ones(4, 1, dtype=torch.bool),
    returns=torch.arange(4.0).reshape(4, 1), prev_values=torch.zeros(4, 1),
    clip_ratio_low=0.2, clip_ratio_high=0.2, value_clip=0.2, huber_delta=10.0,
    entropy_bonus=0.0, entropy=None,
)
policy_loss(values=torch.arange(4.0), **kw)             # squeezed [B]
policy_loss(values=torch.arange(4.0).reshape(4, 1), **kw)  # [B, 1]
```

Both give `critic/value_loss = 1.4649999141693115`.

I checked the other shape combinations the shipped configs use, and none of them reaches a mismatched triple either:

| `reward_type` / `logprob_type` | shapes at the loss |
| --- | --- |
| chunk_level / chunk_level | `(B,)`, `(B,)`, `(B,)` |
| chunk_level / action_level | `(B,1)`, `(B,1)`, `(B,1)` |
| chunk_level / token_level | `(B,1,1)`, `(B,1,1)`, `(B,1,1)` |

Every OpenPI PPO config sets `reward_type: chunk_level`, so the first three rows cover all of them. The `action_level` reward configs use models whose value head width already matches `num_action_chunks` (`mlp_policy` sizes its head by `value_granularity`, `openvla_oft` by `value_type`, and `openvla`/`cnn_policy` run with `num_action_chunks: 1`).

So the `[B]` vs `[B, 1]` difference is real in `rlinf/models/embodiment/openpi/tasks/rl.py:502-509` — it disagrees with both the rollout path a few lines above and with `nft_forward`, which appends `[:, None]` to the same two value sources — but it is cosmetic rather than gradient-affecting. The same squeeze exists in `value_t.mean(dim=-1, keepdim=False)` in GR00T N1.5/N1.6/N1.7, Dexbotic DM0/Pi0, AbotM0 and LingbotVLA, and is equally inert for the same reason.

Happy to send a PR that normalizes those eight sites to `[bsz, 1]` and makes the shape agreement in `compute_ppo_critic_loss` explicit instead of implicit, if that is wanted as hardening. But it would not change any training result, so I did not want to open it under a bug title. If you have a config where the critic loss actually broadcasts, could you post its `reward_type` and `logprob_type`? That would tell us which path I missed.
